stdio/file.c

```diff
@@ -885,8 +885,11 @@ int puts(const char *s)
 
 int fputs_unlocked(const char *s, FILE *stream)
 {
-	int len = strlen(s);
-	return fwrite_unlocked(s, 1, len, stream);
+	size_t len = strlen(s);
+	if (fwrite_unlocked(s, 1, len, stream) != len) {
```

Comment:
I think the write should be retried until either len bytes is written or error is encoutered